.release-notes/4147.md

```diff
@@ -0,0 +1,12 @@
+## change error message
+error message in the situation:
+```
+actor Main
+        let my_test_class: TestClass= TestClass
+        new create(env: Env) =>
+                """placeholder"""
+        fun test() =>
+                my_test_class.x= 1
+```
+was changed from ```not safe to write right side to left side``` to 
+```cannot write to a field in a box function. If you are trying to change state in a function use fun ref.```
```

Comment:
So, I would suggest that you change this to something like showing some example code that leads to the confusing error message from your issue and then what it was changed to.

Think about this as a user of Pony, why would you care that an error message changed? You probably wouldn't. But you'd be interested if a confusing message was improved.

Reply:
sorry i dont understand, could you rephrase?

Reply:
ive updated the message to be more like the format you have described, as for the title i am not very good with words and am still unsure. if you have any specific changes let me know and i will update the file with them. also the error message is only a small inconvenience, so if you think its not worth the effort let me know and i can close the pull request.